lib/queue.rb

```diff
@@ -1,31 +1,57 @@
 class Queue
 
   def initialize
-    # @store = ...
-    raise NotImplementedError, "Not yet implemented"
+    @store = Array.new(100)
+    @front = @back = -1 # nothing has been set yet so both will be set to -1
   end
 
   def enqueue(element)
-    raise NotImplementedError, "Not yet implemented"
+    if @front == -1 && @back == -1
+      @front = 0
+      @back = 0
+    end
+
+    if ((@back + 1) % @store.length) == @front
+      raise StandardError.new
+    end
+
+    @store[@back] = element
+    @back = (@back + 1) % @store.length # but what happens when it reaches the end?
   end
 
   def dequeue
-    raise NotImplementedError, "Not yet implemented"
+    element = @store[@front]
+    @store[@front] = nil
+    @front = (@front + 1) % @store.length
+    return element
```

Comment:
You should also include a check to see if the Queue is empty after removing an element.

```suggestion
    element = @store[@front]
    @store[@front] = nil
    @front = (@front + 1) % @store.length
    @front = @back = -1 if @front == @back  # reset it to empty if front caught up to back
    return element
```